platform/default/mbgl/map/map_snapshotter.cpp

```diff
@@ -44,18 +58,80 @@ MapSnapshotter::Impl::Impl(FileSource& fileSource,
 
     // Set region, if specified
     if (region) {
-        mbgl::EdgeInsets insets = { 0, 0, 0, 0 };
-        std::vector<LatLng> latLngs = { region->southwest(), region->northeast() };
-        map.jumpTo(map.cameraForLatLngs(latLngs, insets));
+        this->setRegion(*region);
     }
 }
 
 void MapSnapshotter::Impl::snapshot(ActorRef<MapSnapshotter::Callback> callback) {
     map.renderStill([this, callback = std::move(callback)] (std::exception_ptr error) mutable {
-        callback.invoke(&MapSnapshotter::Callback::operator(), error, error ? PremultipliedImage() : frontend.readStillImage());
+
+        // Create lambda that captures the current transform state
+        // and can be used to translate for geographic to screen
+        // coordinates
+        assert (frontend.getTransformState());
+        PointForFn pointForFn { [=, center=map.getLatLng(), transformState = *frontend.getTransformState()] (const LatLng& latLng) {
+            LatLng unwrappedLatLng = latLng.wrapped();
+            unwrappedLatLng.unwrapForShortestPath(center);
+            Transform transform { transformState };
+            return transform.latLngToScreenCoordinate(unwrappedLatLng);
+        }};
+
+        // Collect all source attributions
+        std::vector<std::string> attributions;
```

Comment:
By the way, we decided in #6465, we decided that mbgl wouldn’t be responsible for collecting each source’s attribution strings. I take it this is merely a stopgap until we have full runtime styling support for snapshots?

Reply:
The situation is slightly different here as the Map/Style/Sources live on another thread. This means we can't call them directly from the main thread and copying them over just to give access to the attribution strings seems like a waste. That's why I opted to pass them along with the snapshot result.